### PR TITLE
Document Requests kwargs

### PR DESCRIPTION
**Pull request by Kludex**
Should we actually remove the kwargs and use the parameters? It's just a bad experience in every sense to have kwargs.

### CODE REVIEW

File: logfire/__init__.py [hunk 1]
```diff
@@ -150,9 +151,11 @@ def loguru_handler() -> Any:
...
     'suppress_instrumentation',
+    'SamplingOptions',
+    'MetricsOptions',
+    # Integrations
     'StructlogProcessor',
     'LogfireLoggingHandler',
     'loguru_handler',
-    'SamplingOptions',
-    'MetricsOptions',
+    'CeleryInstrumentKwargs',
```
Comment (alexmojaki) [inline logfire/__init__.py]: I think we should in general avoid adding things to `__init__` if they only serve as type hints.

**Review by alexmojaki — CHANGES_REQUESTED** [review]
Normal arguments are definitely better. Each integration can also have `**kwargs: Any` which is documented as something like 'other arguments passed to OTEL, intended for forward compatibility'.

**Comment by Kludex** [comment]
@alexmojaki please confirm if this is okay for you. If it's, I'll add the others.

File: logfire/_internal/main.py [hunk 2]
```diff
@@ -1174,17 +1174,26 @@ def instrument_httpx(
         self._warn_if_not_initialized_for_instrumentation()
         return instrument_httpx(self, client, **kwargs)
 
-    def instrument_celery(self, **kwargs: Unpack[CeleryInstrumentKwargs]) -> None:
+    def instrument_celery(self, **kwargs: Any) -> None:
```
Comment (alexmojaki) [inline logfire/_internal/main.py:1227]: What about `skip_dep_check`?

File: logfire/_internal/main.py [hunk 3]
```diff
@@ -1238,18 +1247,34 @@ def instrument_django(
             **kwargs,
         )
 
-    def instrument_requests(self, excluded_urls: str | None = None, **kwargs: Any) -> None:
+    def instrument_requests(
+        self,
+        excluded_urls: str | None = None,
+        request_hook: Callable[[Span, requests.PreparedRequest], None] | None = None,
```
Comment (alexmojaki) [inline logfire/_internal/main.py:1303]: Nice of otel to pass the actual request object this time lol

**Review by alexmojaki — COMMENTED** [review]
Yes, this approach looks good

File: logfire/_internal/main.py (hunk 2 above)
Comment (Kludex) [reply logfire/_internal/main.py:1227]: I removed on purpose. Is it really important? People shouldn't use it anyway... Unless specific cases like `psycopg-binary`...

**Comment by Kludex** [comment]
let me know if this is okay, and I'll apply to the others.

File: logfire/_internal/main.py (hunk 2 above)
Comment (alexmojaki) [reply logfire/_internal/main.py:1227]: No it's not that important but I wasn't sure if it was intentional. Let's leave it out.

**Pull request merged**